include/hobbes/eval/cc.H

```diff
@@ -172,20 +161,20 @@ public:
   void dumpModule();
 
   // get the x86 machine code for an expression (useful for debugging)
-  typedef std::vector<unsigned char> bytes;
+  using bytes = std::vector<unsigned char>;
   bytes machineCodeForExpr(const std::string& expr);
 
   // keep track of C++ classes so that we can perform upcasts where necessary
   template <typename T>
     void addObj() {
-      hlock _;
+      std::lock_guard<std::recursive_mutex> lk{hccmtx};
```

Comment:
Then we can use the helpers previously defined to introduce a protected computation context:

```c++
template<typename T>
void addObj() {
  return hlock([this](){ return objs->add<T>(); })
}
```